src/highlight/HighlightAnnotation.tsx

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import * as ReactRedux from 'react-redux';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import { getIsCurrentFileVersion } from '../store';
+import { MOUSE_PRIMARY } from '../constants';
+import { Rect } from '../@types/model';
+import './HighlightAnnotation.scss';
+
+type Props = {
+    annotationId: string;
+    className?: string;
+    onSelect?: (annotationId: string) => void;
+    rects: Array<Rect>;
+};
+
+export type HighlightAnnotationRef = HTMLAnchorElement;
+
+const HighlightAnnotation = (props: Props, ref: React.Ref<HighlightAnnotationRef>): JSX.Element => {
+    const { annotationId, className, onSelect = noop, rects } = props;
+    const isCurrentFileVersion = ReactRedux.useSelector(getIsCurrentFileVersion);
+
+    const handleClick = (event: React.MouseEvent<HighlightAnnotationRef>): void => {
+        event.preventDefault();
+        event.stopPropagation();
+        event.nativeEvent.stopImmediatePropagation();
+    };
+
+    const handleFocus = (): void => {
+        onSelect(annotationId);
+    };
+
+    const handleMouseDown = (event: React.MouseEvent<HighlightAnnotationRef>): void => {
+        if (event.buttons !== MOUSE_PRIMARY) {
+            return;
+        }
+
+        onSelect(annotationId);
+
+        event.preventDefault(); // Prevents focus from leaving the anchor immediately in some browsers
+        event.nativeEvent.stopImmediatePropagation(); // Prevents document event handlers from executing
+    };
+
+    return (
+        // eslint-disable-next-line jsx-a11y/anchor-is-valid
+        <a
```

Comment:
Should we use `button` directly? I think anchor for navigation and button is for custom behavior.

Reply:
This is a svg anchor, which allows mouse events on the invisible rects which a html button doesn't let us do